src/app/component/matrix/matrix.component.ts

```diff
@@ -130,26 +153,109 @@ export class MatrixComponent implements OnInit {
   rowCtrl = new FormControl('');
+  rowCtrlTags = new FormControl('');
   filteredRows: Observable<string[]>;
+  filteredActivities: Observable<string[]>;
 
   autoCompeteResults: string[] = [];
+  autoCompleteActivityResults: string[] = [];
 
   @ViewChild('rowInput') rowInput!: ElementRef<HTMLInputElement>;
+  @ViewChild('tagInput') tagInput!: ElementRef<HTMLInputElement>;
+
+  updateActivitesBeingDisplayed(): void {
+    // Iterate over all objects and create new MATRIX_DATA
+    var updatedActivities: any = [];
+
+    for (let i = 0; i < this.allDimensionNames.length; i++) {
+      var subdimensionsInCurrentDimension = Object.keys(
+        this.YamlObject[this.allDimensionNames[i]]
+      );
+
+      for (let j = 0; j < subdimensionsInCurrentDimension.length; j++) {
+        var temp: any = {
+          Dimension: this.allDimensionNames[i],
+          SubDimension: subdimensionsInCurrentDimension[j],
+        };
+        for (let k = 0; k < this.levels.length; k++) {
+          temp = {
+            ...temp,
+            [this.lvlColumn[k] as keyof number]: [],
+          };
+        }
+
+        var activityInCurrentSubDimension: string[] = Object.keys(
+          this.YamlObject[this.allDimensionNames[i]][
+            subdimensionsInCurrentDimension[j]
+          ]
+        );
+
+        for (let a = 0; a < activityInCurrentSubDimension.length; a++) {
+          var currentActivityName = activityInCurrentSubDimension[a];
+          var tagsInCurrentActivity: string[] =
+            this.YamlObject[this.allDimensionNames[i]][
+              subdimensionsInCurrentDimension[j]
+            ][currentActivityName].tags;
+          let flag = 0;
+          if (tagsInCurrentActivity) {
+            for (let curr = 0; curr < tagsInCurrentActivity.length; curr++) {
+              if (
+                this.activityCurrentlyBeingShown.includes(
+                  tagsInCurrentActivity[curr]
+                )
+              ) {
+                flag = 1;
+              }
+            }
+          }
+          if (flag === 1) {
+            try {
+              var lvlOfActivity: number =
+                this.YamlObject[this.allDimensionNames[i]][
+                  subdimensionsInCurrentDimension[j]
+                ][currentActivityName]['level'];
+
+              (
+                temp[
+                  this.lvlColumn[lvlOfActivity - 1] as keyof number
+                ] as unknown as string[]
+              ).push(currentActivityName);
+            } catch {
+              console.log('Level for Activity does not exist');
+            }
+          }
+        }
+        if (this.rowsCurrentlyBeingShown.includes(temp.SubDimension)) {
+          updatedActivities.push(temp);
+        }
+      }
+    }
 
-  //Remove chips
-  remove(row: string): void {
+    this.dataSource.data = JSON.parse(JSON.stringify(updatedActivities));
+  }
+  //Remove
+  // Remove from SubDimension Filter
+  removeSubDimension(row: string): void {
```

Comment:
Please remove the comment above and rename the function to:
removeFromSubDimensionFilter

Like that, you have the description in code and no comment is needed.

Reply:
Okay, I have updated the names